x-pack/plugins/apm/public/components/app/ServiceOverview/view.tsx

```diff
@@ -5,31 +5,78 @@
  */
 
 import { EuiPanel } from '@elastic/eui';
-import React from 'react';
+import { EuiLink } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useEffect } from 'react';
+import chrome from 'ui/chrome';
+import { toastNotifications } from 'ui/notify';
+import url from 'url';
 import { IUrlParams } from 'x-pack/plugins/apm/public/store/urlParams';
 import { useFetcher } from '../../../hooks/useFetcher';
 import { loadServiceList } from '../../../services/rest/apm/services';
-import { loadAgentStatus } from '../../../services/rest/apm/status_check';
 import { NoServicesMessage } from './NoServicesMessage';
 import { ServiceList } from './ServiceList';
 
 interface Props {
   urlParams: IUrlParams;
 }
 
+const initalData = {
+  items: [],
+  hasHistoricalData: true,
+  hasLegacyData: false
+};
+
+let hasDisplayedToast = false;
+
 export function ServiceOverview({ urlParams }: Props) {
   const { start, end, kuery } = urlParams;
-  const { data: agentStatus = true } = useFetcher(() => loadAgentStatus(), []);
-  const { data: serviceListData } = useFetcher(
+  const { data = initalData } = useFetcher(
     () => loadServiceList({ start, end, kuery }),
     [start, end, kuery]
   );
 
+  useEffect(
+    () => {
+      if (data.hasLegacyData && !hasDisplayedToast) {
+        hasDisplayedToast = true;
+        toastNotifications.addWarning({
+          title: i18n.translate('xpack.apm.serviceOverview.toastTitle', {
+            defaultMessage:
+              'Legacy data was detected within the selected time range'
+          }),
+          text: (
+            <p>
+              {i18n.translate('xpack.apm.serviceOverview.toastText', {
+                defaultMessage:
+                  'Your are running the Elastic Stack 7.0+ and incompatible data from a previous 6.x version was detected. If you wish to view this data in APM you must migrate it. See more in '
```

Comment:
Just a few copy tweaks; 

You're running Elastic Stack 7.0+ and we've detected incompatible data from a previous 6.x version. If you want to view this data in APM, you should migrate it. See more in

Reply:
Thanks @formgeist! Will update the copy.